src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,88 @@
+package tasks;
+
+import java.time.LocalDate;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * This class is intended to be extended by more specific task types.
+ */
+
+public class Task {
+    protected String action;
+    protected boolean complete;
```

Comment:
To align with naming conventions, could we rename `complete` to `isComplete`? This would help linters verify that this style rule is being followed. 😄